### PR TITLE
Refactor associations and unassociations, add test coverage, renamed some attributes

### PR DESCRIPTION
**Pull request by rbikar**
This PR is a refactor of creation of units that should associated and unassociated where I added more abstraction which should result better readability and maintainability. 
Renamed **out_repo_set** attr to **repos**, of UbiPopulateRunner class. I found it too long.

Also added tests to improve code coverage.

### CODE REVIEW

File: tests/test_ubipop.py [hunk 1]
```diff
@@ -308,3 +316,137 @@ def test_create_output_file(mock_ubipop_runner, mock_get_repo_pairs, mocked_ubic
...
+    modules, rpms, srpms, debug_rpms = unassociations
+    assert len(modules.units) == 0
+    assert len(rpms.units) == 0
+    assert len(srpms.units) == 0
+    assert len(debug_rpms.units) == 0
+
+
+@pytest.fixture()
+def set_logging():
+    _LOG = logging.getLogger("ubipop")
```
Comment (alexandrevicenzi) [inline tests/test_ubipop.py]: not global, I guess no need to uppercase and _ it.

File: ubipop/__init__.py [hunk 2]
```diff
@@ -221,55 +223,70 @@ def _parse_blacklist_config(self):
...
+    def _determine_pulp_actions(self, units, current, diff_f):
+        if isinstance(units, dict):
+            expected = list(chain.from_iterable(units.values()))
+        else:
+            expected = units
+        to_associate = diff_f(expected, current)
+        to_unassociate = diff_f(current, expected)
+        return to_associate, to_unassociate
+
+    def _gets_action_for_mds(self, modules, current):
```
Comment (rohanpm) [inline ubipop/__init__.py]: Naming here is a bit weird:

- this one is called *gets* action *for* mds
- next one is called *get* action pkgs (no "s", no "for")

Those kind of unnecessary inconsistencies make the code harder to work with.

Comment (rbikar) [reply ubipop/__init__.py]: fixed

File: tests/test_ubipop.py (hunk 1 above)
Comment (rbikar) [reply tests/test_ubipop.py]: fixed

**Pull request merged**